django_mesh/tests/test_models.py

```diff
@@ -44,23 +120,80 @@ def test_render_stays_the_same_when_no_links(self):
         self.p1.save()
 
         self.assertEqual(self.p1.rendered_text, self.p1.text)
+    @patch('requests.sessions.Session.request')
+    def test_automatic_anchors_to_links_on_own_line_if_markdown_chosen(self, mock_request):
+        youtube = get_mock_no_oembed()
+        mock_request.return_value = Mock()
+        mock_request.return_value.status_code = youtube.ysc
+        mock_request.return_value.url = youtube.yu
+        mock_request.return_value.headers = youtube.yh
+        mock_request.return_value.cookies = youtube.yc
+        mock_request.return_value.text = youtube.yt
+
+        self.p7.text = "foo\n   \n http://somelink\n" # generates foo, paragraph, anchored links
+        expected_html = """<p>foo</p>
+                            <p><a href="http://somelink">http://somelink</a></p>"""
+
+        self.assertEqual(self.p7.rendered_text, '')
+
+        self.c1.save()
+        self.p7.channel = self.c1
+        self.p7.save()
 
-    def test_render_adds_anchors(self):
-        self.p1.text = """foo
-            http://somelink
-            http://anotherlink
-        """
+        self.assertHTMLEqual(expected_html, self.p7.rendered_text)
+        self.assertEqual(mock_request.call_count, 1)
+
+    @patch('requests.sessions.Session.request')
+    def test_embed_works_with_markdown(self, mock_request):
+        youtube = get_mock()
+
+        mock_obj_1 = Mock()
+        mock_obj_1.status_code = youtube.ysc
+        mock_obj_1.url = youtube.yu
+        mock_obj_1.headers = youtube.yh
+        mock_obj_1.cookies = youtube.yc
+        mock_obj_1.text = youtube.yt
+
+        mock_obj_2 = Mock()
+        mock_obj_2.status_code = 200
+        mock_obj_2.url = youtube.you
+        mock_obj_2.headers = youtube.yoh
+        mock_obj_2.text = youtube.yot
+        mock_obj_2.json = youtube.yoj
+
+        mock_request.side_effect = iter([mock_obj_1, mock_obj_2])
 
-        self.assertEqual(self.p1.rendered_text, '')
 
         self.c1.save()
-        self.p1.channel = self.c1
-        self.p1.save()
+        self.p7.text = "http://www.youtube.com/watch?v=Uqa8YSxx8Gs"
+        self.p7.channel = self.c1
+        self.p7.save()
+
+        expected_html = """<p><iframe allowfullscreen="" frameborder="0" height="344" 
+                                src="http://www.youtube.com/embed/Uqa8YSxx8Gs?feature=oembed" width="459">
+                            </iframe></p>""" 
+
+        self.assertHTMLEqual(expected_html, self.p7.rendered_text)
+        self.assertEqual(mock_request.call_count, 2)
+
+    @patch('requests.sessions.Session.request')
+    def test_doesnt_embed_with_surrounding_text(self, mock_request):
+        youtube = get_mock_no_oembed()
```

Comment:
Could you replace this with the mock object that does return oembed info? We want to make sure that it's not oembedded because there's surrounding text, and not because it's not returning oembed info.